lib/ide/src/view/text_editor.rs

```diff
@@ -0,0 +1,70 @@
+//! This module contains TextEditor, an UiComponent to edit Enso Modules or Text Files.
+
+use crate::prelude::*;
+
+use basegl::display::object::DisplayObjectOps;
+use basegl::display::shape::text::glyph::font::FontRegistry;
+use basegl::display::shape::text::text_field::TextField;
+use basegl::display::shape::text::text_field::TextFieldProperties;
+use basegl::display::world::*;
+
+use super::ui_component::UiComponent;
+
+use nalgebra::Vector2;
+use nalgebra::Vector4;
+
+/// TextEditor allows us to edit text files or Enso Modules. Extensible code highlighting is
+/// planned to be implemented for it.
+#[derive(Clone,Debug)]
+pub struct TextEditor {
+    text_field : TextField
+}
+
+impl TextEditor {
+    /// Creates a new TextEditor.
+    pub fn new(world:&World) -> Self {
+        let scene     = world.scene();
+        let camera    = scene.camera();
+        let screen    = camera.screen();
+        let mut fonts = FontRegistry::new();
+        let font      = fonts.get_or_load_embedded_font("DejaVuSansMono").unwrap();
+
+        let properties = TextFieldProperties {
+            font,
+            text_size  : 16.0,
+            base_color : Vector4::new(0.0, 0.0, 0.0, 1.0),
```

Comment:
I mentioned this before, but this is not a good way to represent color.
If we can't introduce proper types now, let's at least give this vector value a name, like `opaque_black`. 
But I'd seriously consider at least making a factory function for "colors".

Reply:
I have a suggestion for that.

Reply:
I removed the `Color` I created because another one already exist. `base_color` should be refactored to use the definitions in `core/data/color`.